app/src/main/java/protect/card_locker/MainActivity.kt

```diff
@@ -1,585 +1,607 @@
+                    updateLoyaltyCardList(false)
+                    inputMode.finish()
+                    invalidateOptionsMenu()
                 }
-                return true;
+                return true
             } else if (inputItem.getItemId() == R.id.action_star) {
-                for (LoyaltyCard loyaltyCard : mAdapter.getSelectedItems()) {
-                    Log.d(TAG, "Starring card: " + loyaltyCard.id);
-                    DBHelper.updateLoyaltyCardStarStatus(mDatabase, loyaltyCard.id, 1);
-                    updateLoyaltyCardList(false);
-                    inputMode.finish();
+                for (loyaltyCard in mAdapter!!.getSelectedItems()) {
+                    Log.d(TAG, "Starring card: " + loyaltyCard.id)
+                    DBHelper.updateLoyaltyCardStarStatus(mDatabase, loyaltyCard.id, 1)
+                    updateLoyaltyCardList(false)
+                    inputMode.finish()
                 }
-                return true;
+                return true
             } else if (inputItem.getItemId() == R.id.action_unstar) {
-                for (LoyaltyCard loyaltyCard : mAdapter.getSelectedItems()) {
-                    Log.d(TAG, "Unstarring card: " + loyaltyCard.id);
-                    DBHelper.updateLoyaltyCardStarStatus(mDatabase, loyaltyCard.id, 0);
-                    updateLoyaltyCardList(false);
-                    inputMode.finish();
+                for (loyaltyCard in mAdapter!!.getSelectedItems()) {
+                    Log.d(TAG, "Unstarring card: " + loyaltyCard.id)
+                    DBHelper.updateLoyaltyCardStarStatus(mDatabase, loyaltyCard.id, 0)
+                    updateLoyaltyCardList(false)
+                    inputMode.finish()
                 }
-                return true;
+                return true
             }
 
-            return false;
+            return false
         }
 
-        @Override
-        public void onDestroyActionMode(ActionMode inputMode) {
-            mAdapter.clearSelections();
-            mCurrentActionMode = null;
+        override fun onDestroyActionMode(inputMode: ActionMode?) {
+            mAdapter!!.clearSelections()
+            mCurrentActionMode = null
         }
-    };
+    }
 
-    @Override
-    protected void onCreate(Bundle inputSavedInstanceState) {
-        SplashScreen.installSplashScreen(this);
-        super.onCreate(inputSavedInstanceState);
+    override fun onCreate(inputSavedInstanceState: Bundle?) {
+        installSplashScreen()
+        super.onCreate(inputSavedInstanceState)
 
         // Delete old cache files
         // These could be temporary images for the cropper, temporary images in LoyaltyCard toBundle/writeParcel/ etc.
-        new Thread(() -> {
-            long twentyFourHoursAgo = System.currentTimeMillis() - (1000 * 60 * 60 * 24);
-
-            File[] tempFiles = getCacheDir().listFiles();
+        Thread(Runnable {
```

Comment:
Looking at https://github.com/CatimaLoyalty/Android/blob/2ac36fa4150acc05e1952851308a836980582dfb/app/src/main/java/protect/card_locker/ImportExportActivity.kt#L167 I don't think the `Runnable` is needed